src/app/components/sidebar/filters/price/Price.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from 'react';
+import { useIntl } from 'react-intl';
+import { Checkbox, RangeInput } from 'ui';
+import * as Styled from './Price.styles';
+import { PriceProps } from './Price.type';
+
+export const Price = ({ onInputChange, minPrice, maxPrice }: PriceProps) => {
+  const { formatMessage } = useIntl();
+
+  const [minPriceState, setMinPriceState] = useState(minPrice);
+  const [maxPriceState, setMaxPriceState] = useState(maxPrice);
+
+  useEffect(() => {
+    if (!minPrice) {
+      setMinPriceState(minPrice);
+    }
+  }, [minPrice]);
+
+  useEffect(() => {
+    if (!maxPrice) {
+      setMaxPriceState(maxPrice);
+    }
+  }, [maxPrice]);
+
+  useEffect(() => {
+    const timer = setTimeout(() => onInputChange({ minPrice: minPriceState }), 1000);
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [minPriceState]);
+
+  useEffect(() => {
+    const timer = setTimeout(() => onInputChange({ maxPrice: maxPriceState }), 1000);
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [maxPriceState]);
```

Comment:
Generalnie nieźle, z tym że zauważ, że powielasz dwa razy tę samą funkcjonalność - funkcjonalność debounce.
Można napisać custom hooka `useDebounceState`, która będzie przyjmować argumenty `onDebounceChange` i `time` a zwracać state i setState (w tym przypadku `minPriceState` lub `maxPriceState`). A w środku będą się znajdować `useState` i `useEffect`

Reply:
Ok, dwa pytania:
1. `useDebounceState` powinien jeszcze przyjmować argument `initialValue`, bo inaczej nie będzie to działało jak ktoś wpisze wartość inputa w URL-u
2. Drugi `useEffect` odpowiada za czyszczenie inputa po zamknięciu filtra przez kliknięcie etykiety. Z tego co rozumiem, to co on robi nie kwalifikuje się do funkcjonalności ukrytej pod nazwą `debounce`, ale możnaby go wrzucić do tego hooka i też by działało, a uniknęlibyśmy powielenia kodu w `PricePanel`. Czy w takim wypadku należy go umieścić w hook-u?